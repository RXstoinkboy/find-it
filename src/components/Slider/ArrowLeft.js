import React from 'react';
import styles from './styles/ArrowLeft.module.scss'
import PropTypes from 'prop-types'

export const ArrowLeft = props => {
    return (
        <button className={styles.arrowLeft} onClick={props.prevSlide} />
    );
};

export default ArrowLeft;

ArrowLeft.propTypes = {
    prevSlide: PropTypes.func.isRequired
}