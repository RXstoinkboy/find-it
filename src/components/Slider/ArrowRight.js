import React from 'react';
import styles from './styles/ArrowRight.module.scss'
import PropTypes from 'prop-types'

export const ArrowRight = props => {
    return (
        <button className={styles.arrowRight} onClick={props.nextSlide} />
    );
};

export default ArrowRight;

ArrowRight.propTypes = {
    nextSlide: PropTypes.func.isRequired
}