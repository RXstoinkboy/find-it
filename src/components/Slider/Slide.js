import React from 'react'
import styles from './styles/Slide.module.scss'
import PropTypes from 'prop-types'

export const Slide = props => {
    const inlineStyle = {
        backgroundImage: `url(${props.image.image})`,
        backgroundPosition: 'center',
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat'
    }

    return (
        <>
            <div className={styles.slide} style={inlineStyle}>
                <a href={props.image.url} className={styles.link}>{props.image.name}</a>
            </div>
        </>
    );
};

export default Slide;

Slide.propTypes = {
    image: PropTypes.object.isRequired
}