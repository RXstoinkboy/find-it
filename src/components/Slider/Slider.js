import React, {Component} from 'react'
import {connect} from 'react-redux'
import PropTypes from 'prop-types'
import styles from './styles/Slider.module.scss'

// actions
import {prevSlide} from '../../actions/prevSlide'
import {nextSlide} from '../../actions/nextSlide'

// components
import ArrowRight from './ArrowRight'
import ArrowLeft from './ArrowLeft'
import Slide from './Slide'

export class Slider extends Component {

    nextSlide = () => {
        this.props.nextSlide()
    }

    prevSlide = () => {
        this.props.prevSlide()
    }

    componentDidMount(){
        setInterval(this.nextSlide, 9000);
    }

    render(){
        return (
            <div className={styles.slider}>
                <div 
                    id='slidesWrapper'
                    className={styles.slidesWrapper} 
                    style={{transform: `translateX(${-(this.props.currentImage * this.props.viewportWidth)}px)`}} 
                >
                {/* loop over all slides which are stored in Redux */}
                    {this.props.collectedData.map((item, index) => (
                        <Slide key={index} image={item} />
                    ))}
                </div>
                <ArrowLeft prevSlide={this.prevSlide} />
                <ArrowRight nextSlide={this.nextSlide} />
            </div>
        );
    }
};

const mapStateToProps = state => {
    return{
        collectedData: state.bgImages.collectedData,
        currentImage: state.bgImages.currentImage,
        viewportWidth: state.bgImages.viewportWidth
    }
}

const mapDispatchToProps = {
    prevSlide,
    nextSlide
}

export default connect(mapStateToProps, mapDispatchToProps)(Slider);

Slider.propTypes = {
    collectedData: PropTypes.array.isRequired,
    currentImage: PropTypes.number.isRequired,
    viewportWidth: PropTypes.number.isRequired,
    prevSlide: PropTypes.func.isRequired,
    nextSlide: PropTypes.func.isRequired
}