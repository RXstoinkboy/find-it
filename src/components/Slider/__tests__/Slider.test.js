import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {Slider} from '../Slider'

const mockProps = {
    collectedData: [],
    currentImage: 0,
    viewportWidth: 1000,
    prevSlide: jest.fn(),
    nextSlide: jest.fn()

}

it(`Slider renders correctly`, ()=>{
    const slider = shallow(<Slider {...mockProps}/>)

    expect(toJson(slider)).toMatchSnapshot()
})