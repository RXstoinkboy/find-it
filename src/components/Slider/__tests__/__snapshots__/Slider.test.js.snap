// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<div
  className="slider"
>
  <div
    className="slidesWrapper"
    id="slidesWrapper"
    style={
      Object {
        "transform": "translateX(0px)",
      }
    }
  />
  <ArrowLeft
    prevSlide={[Function]}
  />
  <ArrowRight
    nextSlide={[Function]}
  />
</div>
`;
